docs/06-ReusingTestCode.md

```diff
@@ -0,0 +1,636 @@
+
+<div class="alert alert-info">
+We already know how to create custom login method in AcceptanceTester class. We used actions from standard modules and combined them to make user easier to log in. Helpers allow us to create **new actions** unrelated to standard modules (or using their internals).  
+</div>
+
+
+```php
+<?php
+namespace Helper;
+// here you can define custom functions for FunctionalTester
+
+class Functional extends \Codeception\Module
+{
+}
+?>
+```
+
+As for actions, everything is quite simple. Every action you define is a public function. Write any public method, run the `build` command, and you will see the new function added into the FunctionalTester class.
+
+
+<div class="alert alert-info">
+Public methods prefixed by `_` are treated as hidden and won't be added to your Actor class.
+</div>
+
+Assertions can be a bit tricky. First of all, it's recommended to prefix all your assert actions with `see` or `dontSee`.
+
+Name your assertions like this:
+
+```php
+<?php
+$I->seePageReloaded();
+$I->seeClassIsLoaded($classname);
+$I->dontSeeUserExist($user);
+?>
+```
+And then use them in your tests:
+
+```php
+<?php
+$I->seePageReloaded();
+$I->seeClassIsLoaded('FunctionalTester');
+$I->dontSeeUserExist($user);
+?>
+```
+
+You can define asserts by using assertXXX methods in modules. Not all PHPUnit assert methods are included in modules, but you can use PHPUnit static methods from the `PHPUnit_Framework_Assert` class to leverage all of them.
+
+```php
+<?php
+
+function seeClassExist($class)
+{
+    $this->assertTrue(class_exists($class));
+    // or
+    \PHPUnit_Framework_Assert::assertTrue(class_exists($class));
+}
+?>
+```
+
+In your helpers you can use these assertions:
+
+```php
+<?php
+
+function seeCanCheckEverything($thing)
+{
+    $this->assertTrue(isset($thing), "this thing is set");
+    $this->assertFalse(empty($any), "this thing is not empty");
+    $this->assertNotNull($thing, "this thing is not null");
+    $this->assertContains("world", $thing, "this thing contains 'world'");
+    $this->assertNotContains("bye", $thing, "this thing doesn`t contain 'bye'");
+    $this->assertEquals("hello world", $thing, "this thing is 'Hello world'!");
+    // ...
+}
+?>
+```
+
+### Resolving Collisions
+
+What happens if you have two modules that contain the same named actions?
+Codeception allows you to override actions by changing the module order.
+The action from the second module will be loaded and the action from the first one will be ignored.
+The order of the modules can be defined in the suite config.
+
+However, some of modules may conflict with each others. In order to avoid confusion which module is used in first place, Framework modules, PhpBrowser, and WebDriver can't be used together. `_conflicts` method of module is used to specify which class or interface it conflicts to. Codeception will throw exception if there will be a module enabled which matches the provided criteria.
```

Comment:
may conflict with each other**~~s~~**

which module is used in **the** first place

**The** `_conflicts` method of **a** module

interface it conflicts **~~to~~with**

Codeception will throw **an** exception